src/components/index/StatisticsDisplay.tsx

```diff
@@ -0,0 +1,144 @@
+import { Container } from "@components/commons/Container";
+import { useUnitSuffix } from "@hooks/useUnitSuffix";
+import {
+  StatsData,
+  SupplyData,
+} from "@defichain/whale-api-client/dist/api/stats";
+import { useEffect, useRef, useState } from "react";
+import CountUp from "react-countup";
+import classNames from "classnames";
+import { useTranslation } from "next-i18next";
+import { useWindowDimensions } from "@hooks/useWindowDimensions";
+import { useWhaleApiClient } from "../../layouts/context/WhaleContext";
+import { calculatePercentage } from "../../shared/calculatePercentage";
+
+export function StatsDisplay() {
+  const api = useWhaleApiClient();
+  const [stats, setStats] = useState<StatsData>();
+  const [supply, setSupply] = useState<SupplyData>();
+  const { t } = useTranslation("page-index");
+  const isFirstLoad = useRef(true);
+  const dimensions = useWindowDimensions();
+
+  useEffect(() => {
+    if (isFirstLoad.current) {
+      isFirstLoad.current = false;
+      void api.stats.getSupply().then((supplyItem) => {
+        setSupply(supplyItem);
+      });
+
+      void api.stats.get().then((statsItem) => {
+        setStats(statsItem);
+      });
+    }
+  });
```

Comment:
Will we have to be calling the api constantly?

Reply:
api is called once on load

Reply:
instead of introducing a flag to keep track of state, why not we consider `useEffect(() => { //api logic }, [])`?

Reply:
hmm sorry need further clarification on this cause if we dont keep track of the state everytime a component updates it keeps calling the api with useEffect

Reply:
How does `<StatsDisplay>` if it has no props? Only possibility is when the parent re-renders, but it doesnt happen until a manual page refresh.

And the reason to use `[]` as the dependency is because it will not re-run